index.html

```diff
@@ -539,10 +610,10 @@ <h3>
 did:web:example.com:u:bob
           </pre>
           <p>
-resolves to the DID document at:
+resolves to the DID document at the index of:
           </p>
           <pre class="nohighlight">
-https://example.com/u/bob/did.json
+https://example.com/u/bob/
```

Comment:
`https://example.com/u/bob/` -> `https://example.com/u/bob/index.json`

Adding 1 more arrow here might help better explain the __potenial__ upgrade path for other representations....

IMO, DID Web SHOULD NOT support representations other than JSON, because it makes interop harder and introduces dangerous security issues which are "not worth it".

Reply:
Likewise strongly opposed to anything other than JSON here

Reply:
The proposed change is to resolve the DID to a directory URL, leaving file selection up to the webserver.

The index file could very well be `index.json`, but it could also be `arbitrary.json` if so configured.

The resolvers would request `https://example.com/u/bob/`.

Reply:
> because it makes interop harder and introduces dangerous security issues which are "not worth it".

Citation? Evidence?

Reply:
@msporny  

> IMO, DID Web SHOULD NOT support representations other than JSON, because it makes interop harder and introduces dangerous security issues which are "not worth it".

DID Core defines 2 JSON Formats... `application/did+json` which only requires an `id`.... and `application/did+ld+json` which requires an `id` and `@context`.... if DID web has to support both, in different files, with potentially different key material, service endpoints, etc... and the JSON version wants to just let JSON be JSON and add fields not defined in the context, and the JSON-LD version wants to add terms only defined in a w3id.org hosted context... and a developer needs to understand both representations... thats all a lot harder than just having a single json file on a web server that has an `id` property that is a DID.

JSON-LD should be possible, but it should not be required for DID Web... and I don't think we get any value from having 2 JSON files that are potentially different... other than pain.

The simpler we can make DID web, the better.

Reply:
@OR13 @msporny 

> The simpler we can make DID web, the better.

I absolutely agree that simplicity in the specification is important.

Is there a middle ground here in which the specification makes a variety of representations possible, but does not require any particular representation?

Resolving to directory paths and leaving file selection up to server configuration and client preference assertion is intended to put the choice of DID representation in the hands of each implementation.

There may be a number of alternative representations in the future, and my preference is a specification that need not be revised each time a new one is added. Static filenames ( `did.json` ) have been a feature of this specification from the early days when it was intended only as a single anchor for a given domain. The added flexibility of various representations via directory paths feels like an appropriate extension following the support of multiple DIDs on a single domain.

>  I don't think we get any value from having 2 JSON files that are potentially different...

My intent with this PR is not to require multiple representations, but rather to allow for variety within the representations used by any given implementation. The language indicating that multiple representations must be kept consistent is not intended to imply that multiple representations must exist - only to note they should be consistent if they do exist.

Reply:
I like a lot of the changes in this PR, but I don't see us getting any value from normatively defining support for multiple JSON representations at this time... 

If I could go back and fix DID Core I would, but since its frozen, I think its best that DID Web support the JSON representation that is a super set and only that JSON representation... which is `application/did+json`.. but since that representation will not be understood by any web server... we should allow `application/json` to be returned for any did web document... google and others also do this... messing with accept headers should probably not be added until there is a really good reason, and so far, I am not seeing one.

We have a path to allowing multiple representations, but we should not take that path, because it introduces lots of security issues and is not worth the complexity or attack surface.

Reply:
> JSON-LD should be possible, but it should not be required for DID Web... and I don't think we get any value from having 2 JSON files that are potentially different... other than pain.

Then we have a really deep technical problem here (no machine-verifiable extensibility mechanism), here with no technical solution being proposed. Digital Bazaar will formally object to any such change to `did:web` -- I don't think you guys are thinking this through.

My suggestion is that we say, "yeah, do `application/did+json` ONLY and make the use of "@context" a MUST". Those that want to process it can do so and those that don't... don't. At least in that case, we can tell which properties were meant to be used and which ones are superfluous/unknown. If we don't do that, we have no idea, and that is a huge security vulnerability that Digital Bazaar is not willing to expose themselves to.

The more I think about this, the more I think not allowing proper content negotiation is a terrible idea. I'm starting to sour on this PR given the direction that @OR and @mprorock seem to be proposing... and I have no idea what portions of it that @gribneau is trying to move towards. We'll need to discuss this.

Reply:
Upon re-reading @gribneau's comments, I tend to agree with him. @OR13 and @mprorock -- you guys seem to be trying to make this PR into something it's not. If you want to push your proposal, do it via a separate PR.

Reply:
I can agree with this.

As an aside, I don't consider the IANA Considerations section of DID Core particularly broken. It's quite consistent with long standing best practices and existing standards (outside of the multiple prefix issue). There are certainly scenarios in which dedicated media types will enable desirable client-side functionality.

Nonetheless, we can limit our view here to JSON as the overall media type, and defer the handling of the `Accept` header. This would effectively mean that implementations could use JSON and derivative representations (JSON-LD) at their discretion, but not others. These seem like reasonable limitations for the moment, particularly given that IANA has not yet embraced the DID-specific JSON media types, and that we have a clear path forward for arbitrary future representations outside of JSON that will keep us within established best practices at that time.

Revisions to get there should look something like this:
* Require only `application/json` as the media type when returning DID documents.
* Move references to other media types, DID representations, and content negotiation (`Accept`) to an information section describing the path forward for future work.
* Add appropriate examples for CORS and Cache-Control headers.

This leaves one outstanding issue: full filename paths.

If we limit to the `.json` extension, consistently with the changes above, the logic to handle that case is simple, and it would enable a third case of DIDs as laid out by @dmitrizagidulin here: https://github.com/w3c-ccg/did-method-web/issues/42#issuecomment-966714855

I am not married to the full filename case, but it strikes me as an easy win, and it makes DID Web clearly and apparently web-native, which could be useful in addressing formal objections.

Can we reach consensus on the list above and the full filename issue?

@OR13 @msporny @dmitrizagidulin @mprorock